sdv/utils.py

```diff
@@ -73,23 +41,34 @@ def get_datetime_format(value):
 def is_datetime_type(value):
     """Determine if the input is a datetime type or not.
 
+    If a ``pandas.Series`` or ``list`` is passed, it will return ``True`` if all the values
+    are datetime. Otherwise, it will check if the value is a datetime.
+
+    Note: it will return ``False`` if ``value`` is a string representing
+    a date before the year 1677.
+
     Args:
-        value (pandas.DataFrame, int, str or datetime):
+        value (array-like iterable, int, str or datetime):
             Input to evaluate.
 
     Returns:
         bool:
             True if the input is a datetime type, False if not.
     """
-    if isinstance(value, Iterable) and not isinstance(value, str):
-        value = get_first_non_nan_value(value)
-
-    return (
-        pd.api.types.is_datetime64_any_dtype(value)
-        or isinstance(value, pd.Timestamp)
-        or isinstance(value, datetime)
-        or bool(get_datetime_format([value]))
-    )
+    if isinstance(value, str) or (not isinstance(value, Iterable)):
+        value = cast_to_iterable(value)
+
+    value = pd.Series(value)
+    value = value[~value.isna()]
+    for item in value:
```

Comment:
This should all work fine, but I do think it might increase the time of the metadata validation. Maybe we should still just grab the first item of the series or a subset